packages/swapper/src/swappers/ChainflipSwapper/endpoints.ts

```diff
@@ -0,0 +1,376 @@
+      const cause = error.cause as AxiosError<any, any>
+      throw Error(cause.response!.data.detail)
+    }
+
+    const { data: swapResponse } = maybeSwapResponse.unwrap()
+
+    if (!swapResponse.id) throw Error('missing swap ID')
+
+    tradeQuoteMetadata.set(tradeQuote.id, swapResponse)
+
+    const depositAddress = swapResponse.address!
+    const { assetReference } = fromAssetId(step.sellAsset.assetId)
+    const isTokenSend = isToken(step.sellAsset.assetId)
+
+    const adapter = assertGetEvmChainAdapter(step.sellAsset.chainId)
+
+    const getFeeDataInput: GetFeeDataInput<EvmChainId> = {
+      to: depositAddress,
+      value: step.sellAmountIncludingProtocolFeesCryptoBaseUnit,
+      chainSpecific: {
+        from,
+        contractAddress: isTokenSend ? assetReference : undefined,
+        data: undefined,
+      },
+      sendMax: false,
+    }
+    const feeData = await adapter.getFeeData(getFeeDataInput)
+    const fees = feeData[FeeDataKey.Average]
+
+    if (!isExecutableTradeStep(step)) throw Error('Unable to execute trade step')
+
+    const unsignedTxInput = await adapter.buildSendApiTransaction({
+      to: depositAddress,
+      from,
+      value: step.sellAmountIncludingProtocolFeesCryptoBaseUnit,
+      accountNumber: step.accountNumber,
+      chainSpecific: {
+        gasLimit: fees.chainSpecific.gasLimit,
+        contractAddress: isTokenSend ? assetReference : undefined,
+        ...(supportsEIP1559
+          ? {
+              maxFeePerGas: fees.chainSpecific.maxFeePerGas!,
+              maxPriorityFeePerGas: fees.chainSpecific.maxPriorityFeePerGas!,
+            }
+          : {
+              gasPrice: fees.chainSpecific.gasPrice,
+            }),
+      },
+    })
+
+    return {
+      chainId: Number(fromChainId(chainId).chainReference),
+      data: unsignedTxInput.data,
+      to: unsignedTxInput.to,
+      from,
+      value: unsignedTxInput.value,
+      gasLimit: unsignedTxInput.gasLimit,
+      maxFeePerGas: unsignedTxInput.maxFeePerGas,
+      maxPriorityFeePerGas: unsignedTxInput.maxPriorityFeePerGas,
+      gasPrice: unsignedTxInput.gasPrice,
+    }
+  },
+  getUnsignedUtxoTransaction: async ({
+    tradeQuote,
+    xpub,
+    accountType,
+    assertGetUtxoChainAdapter,
+    config,
+  }: GetUnsignedUtxoTransactionArgs): Promise<BTCSignTx> => {
+    if (!isExecutableTradeQuote(tradeQuote)) throw Error('Unable to execute trade')
+
+    const brokerUrl = config.REACT_APP_CHAINFLIP_API_URL
+    const apiKey = config.REACT_APP_CHAINFLIP_API_KEY
+
+    const step = tradeQuote.steps[0]
+
+    if (!isExecutableTradeStep(step)) throw Error('Unable to execute step')
+
+    const sellChainflipChainKey = `${step.sellAsset.symbol.toLowerCase()}.${
+      chainIdToChainflipNetwork[step.sellAsset.chainId as KnownChainIds]
+    }`
+    const buyChainflipChainKey = `${step.buyAsset.symbol.toLowerCase()}.${
+      chainIdToChainflipNetwork[step.buyAsset.chainId as KnownChainIds]
+    }`
```

Comment:
As discussed previously in https://github.com/shapeshift/web/pull/8049#discussion_r1837434500 and https://github.com/shapeshift/web/pull/8049#discussion_r1837990324, think that's fine for the time being. Mapping doesn't really make sense as it would need to be updated either way, our best bet is codegen for those assets a la THOR gen, but don't think it makes sense for the time being given the low amount of supported assets?

Reply:
gm d097f8c50149b80579acff700bc7f8e2afbb2a41